content/sips/sip-230.md

```diff
@@ -0,0 +1,99 @@
+---
+sip: 230
+title: Universal Circuit Breaker
+network:  Ethereum & Optimism 
+status: Draft
+type: Governance
+author: Daniel Beal (@dbeal-eth)
+created: 2022/04/07
+---
+
+## Simple Summary
+
+<!--"If you can't explain it simply, you don't understand it well enough." Simply describe the outcome the proposed changes intends to achieve. This should be non-technical and accessible to a casual community member.-->
+
+Create a new contract dedicated to providing a base-level of validation for any values provided by the chainlink oracle.
+
+## Abstract
+
+<!--A short (~200 word) description of the proposed change, the abstract should clearly describe the proposed change. This is what *will* be done if the SIP is implemented, not *why* it should be done or *how* it will be done. If the SIP proposes deploying a new contract, write, "We propose to deploy a new contract that will do x".-->
+
+Implementation of SIP-165, and to some extent, futures, have revealed that we need a more self-contained, generic system for chainlink oracle validation is required. To solve this problem,
+a new circuit breaker contract which all queries to chainlink aggregators must run through is created.
+
+For testing purposes, this class can be used to reduce the provisioning complexity of the system in general by being able to set override prices from the
+circuit breaker system itself.
+
+## Motivation
+
+<!--This is where you explain the reasoning behind how you propose to solve the problem. Why did you propose to implement the change in this way, what were the considerations and trade-offs? The rationale fleshes out what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+
+This is a technical debt item which will improve the maintainability and security of the protocol
+
+### Technical Specification
+
+#### `OracleManager`
+
+This is a new contract added to the system which will replace `ExchangeCircuitBreaker`, portions of `ExchangeRates` where aggregators are stored,
+and any other circuit breaker functionality which previously existed will be removed and replaced with a call to this contract.
+
+This contract will include the following external methods:
+
+* `addAggregator(bytes32 id, address target)`: register chainlink compatible oracle from the given target, referenced with the given `id`. Similar to existing `ExchangeRates.addAggregator`
+* `removeAggregator(bytes32 id, address target)`: opposite of `addAggregator`
+* `reset(bytes32 id, uint value)`: If a circuit breaker was triggered, disarm it and initialize it to the given value for last known
```

Comment:
From discussion: this method about resetting the reference value only. So it should probably be named something like `resetComparisonValue` / `resetReferenceValue`. It's also possible to avoid passing the value in, since if the oracle breaker is being reset it is due to it being "ok", in which case the value can be taked from it directly.

It would only need to be passed to set it to a **different** value from last oracle value, which only makes sense if circuit is **not** broken, but should be. In which case the operator should probably suspend the asset instead of using this method.